logos-codegen/src/mir.rs

```diff
@@ -110,9 +110,13 @@ impl TryFrom<Hir> for Mir {
                 }
 
                 let kind = repetition.kind;
+                let is_dot = *repetition.hir == Hir::dot(!repetition.hir.is_always_utf8());
                 let mir = Mir::try_from(*repetition.hir)?;
 
                 match kind {
+                    RepetitionKind::ZeroOrMore | RepetitionKind::OneOrMore if is_dot => {
+                Err(r#"#[regex]: ".+" and ".*" regexes will always match everything, which is most propably not what you want. If you are looking to match everything until a specific character, you should use a capturing group. E.g., use regex r"\([^\}]\)" to match anything in between two parentheses. Read more about that here: https://github.com/maciejhirsz/logos/issues/302#issuecomment-1521342541."#.into())
```

Comment:
Couple nits:

* `regexes will always match everything` -> `patterns will greedily consume the entire source till the end as Logos does not allow for backtracking`
* `you should use a capturing group` -> `you should use a negative character class`
* `r"\([^\}]\)"` -> `r"\([^)]*\)"` (wrong bracket inside, missing repetition, and I don't think you need to escape everything inside character classes)

Could also use a regular (non-raw) string and break this into multiple lines as in:

```rust
Err(
    "first line \
    second line \
    third line"
    .into()
)
```

is the same as

```rust
Err("first line second line third line".into())
```

but more readable.

Reply:
> * `r"\([^\}]\)"` -> `r"\([^)]*\)"` (wrong bracket inside, missing repetition, [you don't need to escape everything inside character classes](https://regex101.com/r/fM5b8P/1)). Alternatively maybe pick a more readable example (if such a thing exists in regex), like: `r"'[^']*'"`

I didn't know, thanks!

I will apply requested changes later today ;-)